src/compose.js

```diff
@@ -12,15 +12,18 @@ var reverse = require('./reverse');
  * @memberOf R
  * @since v0.1.0
  * @category Function
- * @sig ((y -> z), (x -> y), ..., (o -> p), ((a, b, ..., n) -> o)) -> ((a, b, ..., n) -> z)
- * @param {...Function} functions
+ * @sig ((x -> z), ..., (b -> c), ((a1, a2, ...) -> b), (a1, a2, ...)) -> ((a1, a2, ...) -> z)
```

Comment:
I would like to simplify this sig, but I don't think that quite captures it correctly.  There are at least two things that confuse me.  I don't know why there is a switch from an initial `y` to an initial `x`.  And I don't understand what the second `(a1, a2, ...)` is about.

I'm also quite mixed about using the `a1` / `a2` notation.  I would love to be able to get rid of it altogether, as I think it confuses things.  But there seem to be times we need it.

To that end, I've been thinking lately about switching how we document these variadic signatures.  But the ones that are variadic in two ways are harder.  My thought was to express the ellipsis as applying only to a collection of full signatures.  For example,

```js
// compose :: (b -> c) -> (a -> b) -> (a -> c) 
//         :: (c -> d) -> (b -> c) -> (a -> b) -> (a -> d)
//         :: (d -> e) -> (c -> d) -> (b -> c) -> (a -> b) -> (a -> e)
//         :: ...
```

The trouble, of course, is that `compose` is variadic in a second dimension:

```js
// compose :: (b -> c) -> (a -> b) -> (a -> c) 
//         :: (c -> d) -> (a -> b -> c) -> (a -> b -> d)
//         :: (d -> e) -> (a -> b -> c -> d) -> (a -> b -> c -> e)
//         :: ...
```

And while we could certainly write out an example of combining these, there is no way to clearly express this in a linear pattern like those above.

```js
//         :: (f -> g) -> (e -> f) -> (d -> e) -> (a -> b -> c -> d) -> (a -> b -> c -> g)
```

But I'm still tempted, as these lists seem a lot more intuitive than the in-line ellipsis.  I'll open up a separate issue for this, as this is not the place to discuss the notion in detail, but it did seem _a propos_, given this suggested signature change.